### ISSUE TITLE
Instant injectors such as the hypopen are really unfun to play against

### ISSUE THREAD

**Issue by Simyon264** [space-wizards/space-station-14#28659]
## Description

Playing against instant injectors such as the hypopen or the CMOs spray is very unfun. The only counter is hiding, which is very unfun, or guns which not every antag has access to. Other than that there is no counterplay.
**Reproduction**

1. Be any antag
2. Don't have gun
3. CMO hunts you and you cannot do anything to CMO as you get sleeped the moment you enter the CMOs range.

**Additional context**

I propose 2 solutions (I prefer the first one):
- The target must not move in order for the injector to work.
- ~~A small doafter for poison reagents.~~
- Make them not work through hardsuits.

**Comment by deltanedas** [comment]
if it has a doafter you can instantly see if cmo is fake and poisoning someone just by having a doafter or not, and hypos just shouldnt work through (blood-red) hardsuits

**Comment by Simyon264** [comment]
Right, yes. Adding the resistance against hardsuits to the list of solutions

**Comment by slarticodefast** [comment]
I partially addressed the hypospray + chloral hydrate weapon problem in one of my PRs  #27454 (temporarily closed during the freeze). It changes it to make you drowsy like it does in SS13 instead of forcing you to sleep instantly. This way you still have a change to fight back or call for help.

**Comment by Barrett-S** [comment]
CMO should not be hunting you unless you are like actively fighting command or are a nukie, in which case, they quite literally have to be near you and have the accuracy/time for multiple injections (since each one is 5u).

I really don't see the point in nerfing hypo when it's like CMO's one thing to protect themselves, can't be wideswung, and needs multiple injections at point blank. CMO hypo is really only great against slow antags, like jug suit with only melee (which is even less meta after the reflect change).

You know how you counter CMO with hypo? Don't stand still, or just have a gun.

**Comment by deltanedas** [comment]
> accuracy

completely trivial given you can use the right click menu to snipe people with hypos

**Comment by VasilisThePikachu** [comment]
We have had the idea before of them being unable to penetrate hardsuits and it's been basically instantly shot down (#17731)

Making them unable to penetrate hardsuits will just make all heads just start wearing hardsuits.

I don't think it should be a requirement to stand still either. I would actually much rather slarticodefast idea so that chlorar hydrate makes you drowsy first

**Comment by UbaserB** [comment]
This unmerged PR already has chloral hydrate drowsiness #27454

**Comment by SlamBamActionman** [comment]
Is it unreasonable to just make the hypo*sprays* have a 0.5-1 second doafter that gets broken immediately upon movement? 

If you are hypo-ing someone that is willing to be healed they're likely standing still anyways, and you're going to be faster than a syringe regardless when treating people. This stops CMOs from running in and souping someone up with Chloral Hydrate in the middle of a firefight, but still keeps it somewhat viable as an offensive tool if used in conjunction wtih other equipment (slip the shitter with soap, inject inject).

**Comment by Sh18RW** [comment]
I think, it should have a chance to miss if difference between movement vectors is greater than some epsilon, also have DoAfter to inject through hardsuit.
If someone doesn't like this idea, for you: how are low-weight or medical syringe with no problems injecting through high-weight metallic solid hardsuit? It uses to abuse mechanics.
Okay, hypo can do it, but not just in time. It should have delay to go through solid material. In any case, security, captain, engineering, CentCom hardsuits should have protection from momental injection.
There are situations of surprise when agent suddenly jumps from shelter and put to sleep his target. But he could fail by some chance. And it too interesting RP situation.

**Comment by Sh18RW** [comment]
I cannot understand those who think that hypospray should be so buffed. None of the administrators will keep track of who uses hypospray. And the fact that hypo breaks RP environment is a fact.

**Comment by Sh18RW** [comment]
> if someone expects you to hypo them you can't do it
What do you mean?

And what about a chance to miss?

**Comment by SlamBamActionman** [comment]
A random chance to miss a hypo is really not great game design. When you use the hypo you are going for a swift attack that can spell disaster if you mess up; a random chance to cheat you out of that setup would feel terrible.
